src/components/Avatar.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useEffect, useState} from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import {ValueOf} from 'type-fest';
+import useNetwork from '@hooks/useNetwork';
+import * as ReportUtils from '@libs/ReportUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import * as StyleUtils from '@styles/StyleUtils';
+import useTheme from '@styles/themes/useTheme';
+import useThemeStyles from '@styles/useThemeStyles';
+import CONST from '@src/CONST';
+import {AvatarType} from '@src/types/onyx/OnyxCommon';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import Image from './Image';
+
+type AvatarProps = {
+    /** Source for the avatar. Can be a URL or an icon. */
+    source?: AvatarSource;
+
+    /** Extra styles to pass to Image */
+    imageStyles?: ViewStyle[];
+
+    /** Additional styles to pass to Icon */
+    iconAdditionalStyles?: StyleProp<ViewStyle>;
+
+    /** Extra styles to pass to View wrapper */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Set the size of Avatar */
+    size?: ValueOf<typeof CONST.AVATAR_SIZE>;
```

Comment:
```suggestion
    size?: AvatarSizeName;
```

`AvatarSizeName` type is defined in `StyleUtils` (but not exported)